### PR TITLE
NormalizerNFKC: add `unify_katakana_g_sounds` option

### PR DESCRIPTION
**Pull request by HashidaTKS**
When `unify_katakana_g_sounds` is specified, `NormalizerNFKC*` normalize characters as below.

グァ -> ガ
グィ -> ギ
グェ -> ゲ
グォ -> ゴ

Usage: 

```
normalize \
  'NormalizerNFKC130("unify_katakana_g_sounds", true, \
                     "report_source_offset", true)' \
  "グァグィグェグォ" \
  WITH_CHECKS|WITH_TYPES
```

### CODE REVIEW

File: lib/normalizer.c [hunk 1]
```diff
@@ -1718,6 +1796,23 @@ grn_nfkc_normalize_unify(grn_ctx *ctx,
...
+                                      data,
+                                      &unify,
+                                      grn_nfkc_normalize_unify_katakana_g_sounds,
+                                      NULL,
+                                      "[unify][katakana-g-sounds]");
+    if (ctx->rc != GRN_SUCCESS) {
+      goto exit;
+    }
+    need_swap = GRN_TRUE;
+  }
```
Comment (kou) [inline lib/normalizer.c:1814]: Why do you put this between `v_sounds` and `bu_sound`?
`v_sounds` and `bu_sound` are related. `v_sounds` must be processed before `bu_sound` to process `ヴァ` correctly.

If `g_sound` isn't related to `v_sounds` and `bu_sound`, we should not put `g_sound` between `v_sounds` and `bu_sound`.

File: test/command/suite/normalizers/nfkc121/unify_katakana_g_sounds.test [hunk 2]
```diff
@@ -0,0 +1,5 @@
+normalize \
+  'NormalizerNFKC121("unify_katakana_g_sounds", true, \
+                     "report_source_offset", true)' \
+  "グァグィグェグォ" \
```
Comment (kou) [inline test/command/suite/normalizers/nfkc121/unify_katakana_g_sounds.test:4]: Could you add `グ` to check that it's not changed?

```suggestion
  "グァグィググェグォ" \
```

**Comment by kou** [comment]
Could you rebase on master and add a test for `NormalizerNFKC150`?

File: test/command/suite/normalizers/nfkc121/unify_katakana_g_sounds.test (hunk 2 above)
Comment (HashidaTKS) [reply test/command/suite/normalizers/nfkc121/unify_katakana_g_sounds.test]: Added.

File: lib/normalizer.c (hunk 1 above)
Comment (HashidaTKS) [reply lib/normalizer.c:1814]: Fixed. (Just this PR)

File: test/command/suite/normalizers/nfkc100/unify_katakana_g_sounds.reject [hunk 3]
```diff
@@ -0,0 +1,37 @@
+normalize   'NormalizerNFKC100("unify_katakana_g_sounds", true,                      "report_source_offset", true)'   "グァグィグェグォ"   WITH_CHECKS|WITH_TYPES
```
[the remark this answers is not in the document]
Comment (HashidaTKS) [reply test/command/suite/normalizers/nfkc100/unify_katakana_g_sounds.reject]: Sorry, I have overlooked.
Removed.

File: lib/normalizer.c [hunk 4]
```diff
@@ -1418,6 +1418,83 @@ grn_nfkc_normalize_unify_katakana_v_sounds(grn_ctx *ctx,
...
+  char_length = (size_t)grn_charlen_(ctx, current, end, GRN_ENC_UTF8);
+
+  *n_used_bytes = char_length;
+  *n_used_characters = 1;
+
+  /* U+30B0 KATAKANA LETTER GU */
+  if (char_length == 3 &&
+      current[0] == 0xe3 &&
+      current[1] == 0x82 &&
+      current[2] == 0xB0) {
```
[the remark this answers is not in the document]
Comment (HashidaTKS) [reply lib/normalizer.c]: Fixed.

**Comment by HashidaTKS** [comment]
> Could you rebase on master and add a test for NormalizerNFKC150?

Added.

**Comment by HashidaTKS** [comment]
@kou 

Thank you for your comments.
I have addressed your comments.

**Pull request merged**